akd/planner/field_mapping_registry.py

```diff
@@ -0,0 +1,340 @@
+        self,
+        source_agent_id: str,
+        target_agent_id: str
+    ) -> Optional[dict[str, str]]:
+        """
+        Get field mapping for source->target agent pair.
+
+        Priority:
+        1. Explicit mappings (highest trust)
+        2. LLM-generated mappings (if user approved)
+
+        Args:
+            source_agent_id: Source agent identifier
+            target_agent_id: Target agent identifier
+
+        Returns:
+            Dict mapping target_field_name -> source_field_name, or None
+        """
+        key = f"{source_agent_id}->{target_agent_id}"
+
+        # Priority 1: Explicit mappings
+        if key in self.explicit_mappings:
+            logger.debug(f"Using explicit mapping for {key}")
+            return self.explicit_mappings[key]
+
+        # Priority 2: LLM-generated (if approved)
+        if key in self.llm_mappings:
+            entry = self.llm_mappings[key]
+            if entry.user_approved:
+                logger.debug(
+                    f"Using LLM-generated mapping for {key} "
+                    f"(confidence: {entry.confidence:.2f})"
+                )
+                return entry.mapping
+            else:
+                logger.debug(
+                    f"LLM mapping exists for {key} but not user-approved, skipping"
+                )
+
+        return None
+
+    def save_llm_mapping(
+        self,
+        source_agent_id: str,
+        target_agent_id: str,
+        mapping: dict[str, str],
+        confidence: float,
+        user_approved: bool,
+        reasoning: dict[str, str]
+    ):
+        """
+        Save LLM-generated mapping to persistent storage.
+
+        Args:
+            source_agent_id: Source agent identifier
+            target_agent_id: Target agent identifier
+            mapping: Field mappings (target -> source)
+            confidence: Overall confidence score
+            user_approved: Whether user approved the mapping
+            reasoning: Per-field reasoning
+        """
+        key = f"{source_agent_id}->{target_agent_id}"
+
+        entry = LLMMappingEntry(
+            mapping=mapping,
+            generated_at=datetime.now(timezone.utc).isoformat().replace("+00:00", "Z"),
+            confidence=confidence,
+            user_approved=user_approved,
+            reasoning=reasoning
+        )
+
+        self.llm_mappings[key] = entry
+        self._save_llm_file()
+
+        approval_status = "approved" if user_approved else "pending approval"
+        logger.info(
+            f"Saved LLM mapping for {key} "
+            f"(confidence: {confidence:.2f}, {approval_status})"
+        )
+
+    def _save_llm_file(self):
+        """Persist LLM mappings to JSON file."""
+        try:
+            # Ensure directory exists
+            path = Path(self.llm_path)
+            path.parent.mkdir(parents=True, exist_ok=True)
+
+            # Convert to dict for JSON serialization
+            data = {
+                "version": "1.0.0",
```

Comment:
Need this to be switched to `WORKFLOW_FORMAT_VERSION` for flexibility in newer versioning maybe?